### DISCUSSION TITLE
How you execute the cli package?

### DISCUSSION THREAD

**Discussion by jon9090** [coveo/cli#836] [Q&A] [answered]
This is not bug. but a repo question. 

How do you execute the `cli` package in development? 

For each change in the source code you build the package and then execute it from the dist? or you using nx command to execute the code?

I know that the entry file is  https://github.com/coveo/cli/blob/master/packages/cli/bin/run
But I am not sure how you execute this file.

**Comment by louis-bompart** [comment, marked as answer]
Hi @jon9090,

In development, we use the https://github.com/coveo/cli/blob/master/packages/cli/bin/dev file. You can see from the shebang that this file is meant to be executed with NodeJS.

This is however only valid for the `@coveo/cli` packages. This repo uses a mono repo architecture, so there are other projects relative to the CLI here. Still, those changes are not directly bundled with the CLI, but instead published on npm, for example, our angular schematic, `@coveo/angular`.

So, to test 'the full loop' (for changes outside the `packages/cli`), we use Verdaccio to start a local npm registry where we can publish the changed version.

This is what we do in our End-to-End tests.

I hope this answer your question :)
